experimental/FTheoryTools/src/TateModels/methods.jl

```diff
@@ -95,66 +94,84 @@ Construction over concrete base may lead to singularity enhancement. Consider co
 
 Global Tate model over a concrete base -- SU(5)xU(1) restricted Tate model based on arXiv paper 1109.3454 Eq. (3.1)
 
-julia> blow_up(t, ["x", "y", "x1"]; coordinate_name = "e1")
-Partially resolved global Tate model over a concrete base -- SU(5)xU(1) restricted Tate model based on arXiv paper 1109.3454 Eq. (3.1)
+julia> x1, x2, x3, x4 = gens(cox_ring(base_space(t)))
+4-element Vector{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}:
+ x1
+ x2
+ x3
+ x4
+
+julia> my_choice = Dict("a1" => x1^4, "a2" => x1^8, "w" => x2 - x3)
+Dict{String, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}} with 3 entries:
+  "w"  => x2 - x3
+  "a2" => x1^8
+  "a1" => x1^4
+
+julia> tuned_t = tune(t, my_choice)
+Global Tate model over a concrete base
+
+julia> tate_section_a1(tuned_t) == x1^4
+true
 ```
 """
-function blow_up(t::GlobalTateModel, ideal_gens::Vector{String}; coordinate_name::String = "e")
-  R = cox_ring(ambient_space(t))
-  I = ideal([eval_poly(k, R) for k in ideal_gens])
-  return blow_up(t, I; coordinate_name = coordinate_name)
-end
+function tune(t::GlobalTateModel, input_sections::Dict{String, <:Any}; completeness_check::Bool = true)
+  # Consistency checks
+  @req base_space(t) isa NormalToricVariety "Currently, tuning is only supported for models over concrete toric bases"
+  isempty(input_sections) && return t
+  secs_names = collect(keys(explicit_model_sections(t)))
+  tuned_secs_names = collect(keys(input_sections))
+  @req all(x -> x in secs_names, tuned_secs_names) "Provided section name not recognized"
+
+  # 0. Prepare for computation by setting up some information
+  explicit_secs = deepcopy(explicit_model_sections(t))
+  def_secs_param = deepcopy(defining_section_parametrization(t))
+  tate_sections = ["a1", "a2", "a3", "a4", "a6"]
+
+  # 1. Tune model sections different from Tate sections
+  for x in setdiff(tuned_secs_names, tate_sections)
+    section_parent = parent(input_sections[x])
+    section_degree = degree(input_sections[x])
+    @req section_parent == parent(explicit_model_sections(t)[x]) "Parent mismatch between given and existing model section"
+    @req section_degree == degree(explicit_model_sections(t)[x]) "Degree mismatch between given and existing model section"
```

Comment:
The tuning is currently limited to toric bases. In this case, the `explicit_model_sections` are homogeneous polynomials in the Cox ring of the toric base. `degree` computes the multidegree of these polynomials. This will ensure that the sections in questions are sections of the same line bundle.

There are issues once this is being extended to families of base spaces. This is to be discussed in a separate PR. Here, I simply raise an error if we try to tune a model whose base space is not a toric variety.